src/hera/workflow.py

```diff
@@ -320,3 +328,12 @@ def get_parameter(self, name: str) -> Parameter:
         if self.parameters is None or next((p for p in self.parameters if p.name == name), None) is None:
             raise KeyError(f"`{name}` is not a valid workflow parameter")
         return Parameter(name, value=f"{{{{workflow.parameters.{name}}}}}")
+
+    def to_dict(self) -> dict:
+        """Returns the JSON/dictionary representation of the workflow"""
+        return self.build().to_dict()
+
+    def to_yaml(self) -> str:
+        """Returns a YAML representation of the workflow"""
+        dict_repr = model_to_dict(self.build())
+        return yaml.dump(dict_repr)
```

Comment:
Should we instead return json? That way we won't need a dependency on pyyaml and all json documents can be submitted to k8s directly anyway (since json is a strict subset of yaml?)

Reply:
If users need it they can dump the output json to yaml on their own?

Reply:
Personally, I am a fan of the idea of not having the YAML dependency (I avoided adding YAML support to Hera for a while now) but issues like [this one](https://github.com/argoproj-labs/hera-workflows/issues/316) have surfaced before as well. I think there _is_ community appetite to use Hera in a more GitOps fashion, which is the primary factor that influenced the decision to accept the introduction of the YAML dependency. However, I certainly think there's immense value to optionality and access to "control knobs" that dictate whether one installs a dependency or not, so I will update the dependencies to make YAML optional. I have a hunch that the part of our community that does use GitOps + Hera likely _already_ has the YAML dependency 

@valorl would love to hear what you think of this!

Reply:
For our usage, I don't think it's very likely that pyyaml would already be installed for other purposes. So from that perspective, it would be certainly convenient. But I can understand not wanting the dependency. I would always want YAML in my GitOps, but that's my preference so I'm fine with having the dependency on my end. Not sure how generally popular that preference is.